src/libraries/Microsoft.VisualBasic.Core/src/Microsoft/VisualBasic/FileSystem.vb

```diff
@@ -1000,6 +1000,7 @@ Namespace Microsoft.VisualBasic
             End Try
         End Sub
 
+        <UnsupportedOSPlatform("macos")> 
         Public Function InputString(ByVal FileNumber As Integer, ByVal CharCount As Integer) As String
```

Comment:
I wonder if this API's area owner would prefer to reduce the aggressiveness of marking the whole platform as unsupported, and instead only calling `oFile.Lock()` and `oFile.Unlock()` when the platform is not MacOS.

Reply:
As far as I know, mono team has a [story ](https://github.com/dotnet/runtime/issues/45193) for adding more attributes for mac/android/ios, they might want to see this cc @marek-safar

Reply:
That story is for SDKs but we are going to audit libraries as well during #47910 work